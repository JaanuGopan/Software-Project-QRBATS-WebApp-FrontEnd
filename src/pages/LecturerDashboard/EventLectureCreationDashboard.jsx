import React from 'react';
import './LectureDashboard.css';
import EventLectureCreation from './EventLectureCreation';
const EventLectureCreationDashboard = () => {
  return (
    <div className="row m-5 lecture-creation-dashboard">
      <EventLectureCreation
        handleCloseCreateLectureWindow={() => null}
        reloadLectureList={() => null}
        hideCloseButton={true}
        showImage={true}
      />
    </div>
  );
};

export default EventLectureCreationDashboard;
