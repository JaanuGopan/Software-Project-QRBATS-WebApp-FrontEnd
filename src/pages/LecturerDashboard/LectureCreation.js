import React, { useState, useEffect, useRef } from "react";
import EventService from "../../api/services/EventService";
import QRCode from "qrcode.react";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";
import "../Event/EventCreation/EventCreation.css";
import eventCreationImage from "../../assets/Images/designer_pic/Designer_pic4.jpeg";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import DualButtonComponent from "../../components/buttons/DualButtonComponent";
import Select from "react-select";
import ModuleService from "../../api/services/ModuleService";
import LocationService from "../../api/services/LocationService";

const LectureCreation = ({
  handleCloseCreateLectureWindow,
  reloadLectureList,
  hideCloseButton,
  locationNameList,
  showImage = false,
}) => {
  const user = useSelector(selectUser);
  const { userId, departmentId } = user || {};

  const [eventId, setEventId] = useState("");
  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState("");
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventEndTime, setEventEndTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState("LECTURE");
  const [eventAssignedUserId, setEventAssignedUserId] = useState(userId);
  const [title, setTitle] = useState("Lecture");
  const [moduleNameList, setModuleNameList] = useState([]);

  const [venueList, setVenuesList] = useState([
    "NCC",
    "LT1",
    "LT2",
    "Auditorium",
    "DEIE",
    "DMME",
    "DCEE",
  ]);

  const handleGetModulesList = async () => {
    const response = await ModuleService.getAllModulesByDepartmentId(
      departmentId
    );
    if (response) {
      const moduleList = response.map((module) => ({
        value: module.moduleId,
        label: module.moduleCode,
      }));
      setModuleNameList(moduleList);
      console.log(moduleList);
    }
  };

  const handleGetLocationNameList = async () => {
    try {
      const response = await LocationService.getAllLocationNames();
      setVenuesList(response);
    } catch (error) {
      console.error("Error fetching location names:", error);
    }
  };

  const locationList = [
    "NCC",
    "LT1",
    "LT2",
    "Auditorium",
    "DEIE",
    "DMME",
    "DCEE",
  ];

  const qrCodeRef = useRef(null);

  const notifySuccess = () => toast.success("Successfully Lecture Created!");

  const clearEventDetails = () => {
    setEventName("");
    setEventDate("");
    setEventValidDate("");
    setEventTime("");
    setEventEndTime("");
    setEventVenue("");
    setEventAssignedUserId(null);
    setEventRole("LECTURE");
    setModuleName(null);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await EventService.saveEvent(
        eventName,
        eventDate,
        eventValidDate,
        eventTime,
        eventEndTime,
        eventVenue,
        eventRole,
        eventRole === "LECTURE" ? moduleName.label : "",
        userId
      );
      setEventId(response.eventId);
      notifySuccess();
      reloadLectureList();
      clearEventDetails();
    } catch (error) {
      console.error("Lecture creation failed", error);
    } finally {
      setShowQRCode(true);
    }
  };

  const downloadQRCode = () => {
    const qrCodeURL = document
      .getElementById("qrCodeEl")
      .toDataURL("image/png")
      .replace("image/png", "image/octet-stream");
    let aEl = document.createElement("a");
    aEl.href = qrCodeURL;
    aEl.download = "QR_Code.png";
    document.body.appendChild(aEl);
    aEl.click();
    document.body.removeChild(aEl);
  };

  const eventDetails = {
    eventId: eventId,
    eventName: eventName,
    moduleName: moduleName,
    eventDate: eventDate,
    eventValidDate: eventValidDate,
    eventTime: eventTime,
    eventEndTime: eventEndTime,
    eventVenue: eventVenue,
    eventAssignedUserId: userId,
  };

  const qrCodeDetails = JSON.stringify(eventDetails);

  const [qrCodeWindow, setQrCodeWindow] = useState(false);

  const [selectedButton, setSelectedButton] = useState(1);

  const handleSelectedButton = (button) => {
    setSelectedButton(button);
    if (button === 2) {
      setEventRole("EVENT");
      setModuleName(null);
      setTitle("Event");
    } else {
      setEventRole("LECTURE");
      setTitle("Lecture");
    }
  };

  const today = new Date().toISOString().split("T")[0];

  const handleCloseQrCodeWindow = () => {
    setShowQRCode(false);
    clearEventDetails();
  };

  useEffect(() => {
    handleGetModulesList();
    handleGetLocationNameList();
    //setUserId(UserDetails.getUserId());
  }, []);

  return (
    <div className="event-main-container1">
      <Toaster />
      {!hideCloseButton && (
        <div
          className="closeCreateEventWindow"
          onClick={handleCloseCreateLectureWindow}
        >
          <IoMdCloseCircleOutline size={25} />
        </div>
      )}
      <h2>Create {` ${title}`}</h2>
      <div className="eventCreation-field">
        {showImage && (
          <img src={eventCreationImage} className="Create-logo" alt="Logo" />
        )}
        <div className="eventCreation-input-field">
          <DualButtonComponent
            onSelect={handleSelectedButton}
            buttonText1={"Lecture"}
            buttonText2={"Event"}
          />
          <form onSubmit={handleSubmit}>
            <div className="input-with-icon">
              <label className="date-label" htmlFor="moduleCode">
                {`${title} Name`}
              </label>
              <input
                required
                type="text"
                id="eventName"
                name="eventName"
                placeholder={`${title} Name`}
                className="form-control mb-2"
                value={eventName}
                onChange={(e) => setEventName(e.target.value)}
              />
            </div>

            {selectedButton === 1 && (
              <div className="input-with-icon">
                <label className="date-label" htmlFor="moduleCode">
                  {`Module Code`}
                </label>
                <Select
                  required
                  id="moduleName"
                  name="moduleName"
                  options={moduleNameList}
                  onChange={(e) => setModuleName(e)}
                  value={moduleName}
                  placeholder={"Select Module Code"}
                />
              </div>
            )}
            <div className="date-div">
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    {`${title} Starting Date`}
                  </label>
                  <input
                    required
                    type="date"
                    min={today}
                    value={eventDate}
                    onChange={(e) => {
                      setEventDate(e.target.value);
                      setEventValidDate(e.target.value);
                    }}
                    className="form-control mb-2"
                  />
                </div>
              </div>
              {/* <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    {`${title} Ending Date`}
                  </label>
                  <input
                    required
                    type="date"
                    min={today}
                    value={eventValidDate}
                    onChange={(e) => setEventValidDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div> */}
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="date-label" htmlFor="eventDate">
                  Venue
                </label>
                <select
                  required
                  value={eventVenue}
                  onChange={(e) => setEventVenue(e.target.value)}
                  className="form-control mb-2"
                >
                  <option value="">Select Venue</option>
                  {venueList.map((option, index) => (
                    <option key={index} value={option.locationName}>
                      {option.locationName}
                    </option>
                  ))}
                </select>
              </div>
            </div>

            <div className="eventCreation-form">
              <div>
                <label className="form-label" htmlFor="eventDate">
                  {`${title} Starting Time`}
                </label>
                <input
                  required
                  type="time"
                  value={eventTime}
                  onChange={(e) => setEventTime(e.target.value)}
                  placeholder={`${title} Time`}
                  className="form-control mb-2"
                />
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="form-label" htmlFor="eventEndTime">
                    {`${title} End Time`}
                  </label>
                  <input
                    required
                    type="time"
                    value={eventEndTime}
                    onChange={(e) => setEventEndTime(e.target.value)}
                    placeholder={`${title} End Time`}
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>

            <div className="eventCreation-form"></div>
            <button
              onClick={() => setQrCodeWindow(true)}
              type="submit"
              className="btn btn-success w-100"
            >
              Create {` ${title}`}
            </button>
          </form>
        </div>
        {qrCodeWindow && showQRCode && (
          <div className="Admin-Create-Event-Dashboard">
            <div ref={qrCodeRef} className="event-main-container1">
              <div
                className="closeCreateEventWindow"
                onClick={handleCloseQrCodeWindow}
              >
                <IoMdCloseCircleOutline />
              </div>
              <h2>Successfully Event Created</h2>
              <div className="row-center">
                <QRCode
                  name="QRCode"
                  size={200}
                  id="qrCodeEl"
                  value={qrCodeDetails}
                  className="mb-2"
                />
              </div>
              <div>
                <p className="text-center">Scan this QR code to join {title}</p>
              </div>
              <div className="row-center">
                <div className="QRbutton">
                  <button
                    onClick={downloadQRCode}
                    className="btn btn-success mr-3"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default LectureCreation;
