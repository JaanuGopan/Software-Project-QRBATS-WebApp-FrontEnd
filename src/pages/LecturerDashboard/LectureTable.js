import React, { useEffect, useState } from "react";
import "../AdminDashboard/AdminDashboard.css";
import { FaEdit } from "react-icons/fa";

const LectureTable = ({
  lectureList,
  onLectureClick,
  search,
  handleUpdateLecture,
}) => {
  const [lecturesList, setLecturesList] = useState([]);
  const [selectedLecture, setSelectedLecture] = useState(null);

  useEffect(() => {
    setLecturesList(lectureList);
  }, [lectureList]);

  const handleLectureClick = (clickedLecture) => {
    onLectureClick(clickedLecture);
    setSelectedLecture(clickedLecture);
  };

  return (
    <div className="tableDesign">
      <table className="tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>Date</th>
            <th>Venue</th>
            <th>Start Time</th>
            <th>End Time</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {lecturesList
            .filter(
              (lecture) =>
                lecture.eventName
                  .toLowerCase()
                  .includes(search.toLowerCase()) ||
                lecture.eventVenue.toLowerCase().includes(search.toLowerCase())
            )
            .map((lecture, index) => (
              <tr
                key={index}
                onClick={() => handleLectureClick(lecture)}
                className={
                  selectedLecture === lecture ? "selected-row" : "event-row"
                }
              >
                <td>{index + 1}</td>
                <td>{lecture.eventName}</td>
                <td>{lecture.eventDate}</td>
                <td>{lecture.eventVenue}</td>
                <td>{lecture.eventTime}</td>
                <td>{lecture.eventEndTime}</td>
                <td>
                  <button
                    onClick={() => handleUpdateLecture(lecture)}
                    className="EditButton"
                  >
                    <FaEdit className="EditIcon" />
                  </button>
                </td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};
export default LectureTable;
