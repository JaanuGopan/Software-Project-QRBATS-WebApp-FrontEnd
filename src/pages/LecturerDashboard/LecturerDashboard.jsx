import React, { useState, useEffect } from 'react';
import '../AdminDashboard/AdminDashboard.css';
import './LectureDashboard.css';
import NormalButton from '../../components/layout/AdminDashboardComponent/NormalButton';
import { MdCreateNewFolder } from 'react-icons/md';
import { RiDeleteBin5Fill } from 'react-icons/ri';
import EventLectureCreation from './EventLectureCreation';
import EventService from '../../api/services/EventService';
import { useSelector } from 'react-redux';
import { selectUser } from '../../redux/features/userSlice';
import AdminUpdateEvent from '../Event/AdminUpdateEvent';
import LocationService from '../../api/services/LocationService';
import LectureService from '../../api/services/LectureService';
import LecturesTable from '../Lectures/LecturesTable';
import LecturesEdit from '../Lectures/LecturesEdit';
import LectureCreationPage from '../LactureCreation/LectureCreationPage';
import { ToastContainer, toast } from 'react-toastify';
import WarningPopup from '../../components/warningPopup/WarningPopup';
import { Table, ToggleButton, ToggleButtonGroup } from '@mui/material';
import EventLectureTable from './EventLectureTable';
const LecturerDashboard = () => {
  const [showCreateLecturePopup, setShowCreateLecturePopup] = useState(false);
  const [showUpdateLecturePopup, setShowUpdateLecturePopup] = useState(false);
  const [showDeleteLecturePopup, setShowDeleteLecturePopup] = useState(false);
  const [showDeleteEventPopup, setShowDeleteEventPopup] = useState(false);
  const [eventLectureList, setEventLectureList] = useState([]);
  const [selectedLecture, setSelectedLecture] = useState(null);
  const [selectedEventLecture, setSelectedEventLecture] = useState(null);
  const [searchLecture, setSearchLecture] = useState('');
  const [selectTable, setSelectTable] = useState('Lectures');
  const [title, setTitle] = useState('Lecture');
  const [venuesList, setVenuesList] = useState([]);
  const user = useSelector(selectUser);
  const { userId } = user || {};

  const [lectureList, setLectureList] = useState([]);

  const handleReloadEventLectureList = async () => {
    EventService.getEventByUserID(userId)
      .then((list) => {
        setEventLectureList(list);
        console.log(eventLectureList);
      })
      .catch((error) => {
        console.log('Error in getting lecture list ', error);
      })
      .finally(() => {});
  };
  //=====================================================================================
  const [showEditLectureWindow, setShowEditLectureWindow] = useState(false);
  const [editLecture, setEditLecture] = useState(null);

  const handleReloadLectureList = async () => {
    const response = await LectureService.getAllLecturesByUserId(userId);
    if (response) {
      setLectureList(response);
      console.log('lecture list : ', response);
    }
  };

  const handleEditLecture = (lecture) => {
    if (lecture) {
      setEditLecture(lecture);
      setShowEditLectureWindow(true);
    }
  };

  const handleShowDeleteWindow = () => {
    if (selectedLecture === null) {
      toast.error('Please Select Lecture To Delete.');
      return;
    }
    setShowDeleteLecturePopup(true);
  };

  const [processingDeleteLecture, setProcessingDeleteLecture] = useState(false);

  const handleDeleteLecture = async () => {
    try {
      setProcessingDeleteLecture(true);
      const response = await LectureService.deleteLecture(selectedLecture.lectureId);
      if (response.status === 200) {
        handleReloadLectureList();
        setSelectedLecture(null);
        toast.success(`Successfully ${selectedLecture.lectureName} Lecture Deleted.`);
        setShowDeleteLecturePopup(false);
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error(`Error In Deleting Lecture ${selectedLecture.lectureName}.`);
      }
    } finally {
      setProcessingDeleteLecture(false);
    }
  };

  //=====================================================================================

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    handleReloadEventLectureList();
    handleGetLocationNameList();
    handleReloadLectureList();
  }, []);

  const handleShowDeleteEventWindow = () => {
    if (selectedEventLecture === null) {
      toast.error('Please Select Event To Delete.');
      return;
    }
    setShowDeleteEventPopup(true);
  };

  const [processingDeleteEvent, setProcessingDeleteEvent] = useState(false);
  const handleDeleteEventLecture = async () => {
    if (selectedEventLecture) {
      try {
        setProcessingDeleteEvent(true);
        const response = await EventService.deleteEvent(selectedEventLecture.eventId);
        if (response.status === 200) {
          handleReloadEventLectureList();
          setSelectedEventLecture(null);
          toast.success(`Successfully ${selectedEventLecture.eventName} Event Deleted.`);
          setShowDeleteEventPopup(false);
        } else if (response.status === 400) {
          toast.error(response.data);
        } else {
          toast.error(`Error In Deleting Lecture ${selectedEventLecture.eventName}.`);
        }
      } finally {
        setProcessingDeleteEvent(false);
      }
    }
  };

  const handleEventLectureClick = (lecture) => {
    setSelectedEventLecture(lecture);
  };

  const handleChange = (e) => {
    setSelectTable(e);
  };

  return (
    <div className="admin-Dash">
      <ToastContainer />
      <p className="mainHead">{'Lecturer Dashboard'}</p>

      <div className="lecturer-dashboard-toggle-buttons-search">
        <div className="SearchEvent">
          <div className="lecturer-dashboard-toggle">
            <label className="lecturer-dashboard-toggle-label">Select Table : </label>
            <div className="size-box"></div>
            <ToggleButtonGroup
              color="primary"
              value={selectTable}
              exclusive
              onChange={(e) => handleChange(e.target.value)}
              aria-label="Platform"
            >
              <ToggleButton value="Lectures">Lectures</ToggleButton>
              <ToggleButton value="Events">Events</ToggleButton>
            </ToggleButtonGroup>
          </div>
          <input
            type="text"
            placeholder="Search..."
            style={{
              width: '150px',
              padding: '3px 40px',
              border: '0.5px solid black',
              borderRadius: '5px',
              textAlign: 'center',
            }}
            onChange={(e) => setSearchLecture(e.target.value)}
          />
        </div>
      </div>
      {selectTable === 'Lectures' ? (
        <div className="lecture-dashboard-container">
          <div className="lecture-dashboard-table">
            <LecturesTable
              lecturesList={lectureList}
              search={searchLecture}
              onLectureClick={(e) => {
                setSelectedLecture(e);
              }}
              handleLectureUpdate={(lecture) => {
                handleEditLecture(lecture);
              }}
            />
          </div>

          <div className="List-Buttons">
            <NormalButton
              handleClick={() => setShowCreateLecturePopup(true)}
              title={'Create Lecture'}
              titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
            />
            <NormalButton
              title={'Delete'}
              handleClick={handleShowDeleteWindow}
              titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
            />
          </div>
        </div>
      ) : (
        <div className="lecture-dashboard-container">
          <div className="lecture-dashboard-table">
            <EventLectureTable
              search={searchLecture}
              handleUpdateLecture={() => setShowUpdateLecturePopup(true)}
              onLectureClick={handleEventLectureClick}
              lectureList={eventLectureList} // Pass the eventList prop here
            />
          </div>
          <div className="List-Buttons">
            <NormalButton
              handleClick={() => setShowCreateLecturePopup(true)}
              title={'Create Event'}
              titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
            />
            <NormalButton
              title={'Delete'}
              handleClick={handleShowDeleteEventWindow}
              titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
            />
          </div>
        </div>
      )}
      {showCreateLecturePopup && (
        <div className="Create-Lecture-Window">
          {selectTable === 'Events' && (
            <EventLectureCreation
              handleCloseCreateLectureWindow={() => setShowCreateLecturePopup(false)}
              reloadLectureList={handleReloadEventLectureList}
              hideCloseButton={false}
              locationNameList={venuesList}
            />
          )}
          {selectTable === 'Lectures' && (
            <LectureCreationPage
              handleCloseCreateLectureWindow={() => setShowCreateLecturePopup(false)}
              handleReloadLectureList={handleReloadLectureList}
              hideCloseButton={false}
            />
          )}
        </div>
      )}
      {showUpdateLecturePopup && (
        <div className="Admin-Create-Event-Dashboard">
          <AdminUpdateEvent
            handleCloseEventUpdateWindow={() => setShowUpdateLecturePopup(false)}
            selectedEvent={selectedEventLecture}
            reloadEventList={handleReloadEventLectureList}
            locationNameList={venuesList}
          />
        </div>
      )}
      {showEditLectureWindow && (
        <div className="Admin-Create-Event-Dashboard">
          <LecturesEdit
            selectedLecture={editLecture}
            handleCloseUpdateLectureWindow={() => {
              setShowEditLectureWindow(false);
              setEditLecture(null);
            }}
            handleReload={handleReloadLectureList}
            locationNameList={venuesList}
            reloadLecturesList={handleReloadLectureList}
          />
        </div>
      )}
      {showDeleteLecturePopup && (
        <div className="delete-lecture-container">
          <WarningPopup
            handleCloseWarningWindow={() => setShowDeleteLecturePopup(false)}
            handleOk={handleDeleteLecture}
            titleText={'Are You Sure You Want To Delete This Lecture?'}
            buttonText={'Delete'}
            processing={processingDeleteLecture}
          />
        </div>
      )}
      {showDeleteEventPopup && (
        <div className="delete-lecture-container">
          <WarningPopup
            handleCloseWarningWindow={() => setShowDeleteEventPopup(false)}
            handleOk={handleDeleteEventLecture}
            titleText={'Are You Sure You Want To Delete This Event?'}
            buttonText={'Delete'}
            processing={processingDeleteEvent}
          />
        </div>
      )}
    </div>
  );
};

export default LecturerDashboard;
