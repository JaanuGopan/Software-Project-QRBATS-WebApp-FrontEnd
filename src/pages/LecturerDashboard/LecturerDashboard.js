import React, { useState, useEffect } from "react";
import "../AdminDashboard/AdminDashboard.css";
import "./LectureDashboard.css";
import LectureTable from "./LectureTable";
import TotalCount from "../../components/layout/AdminDashboardComponent/TotalCount";
import { FaSchool } from "react-icons/fa6";
import { IoNewspaperSharp } from "react-icons/io5";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import LectureCreation from "./LectureCreation";
import EventService from "../../api/services/EventService";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import AdminUpdateEvent from "../Event/AdminUpdateEvent";
import LocationService from "../../api/services/LocationService";
import LectureService from "../../api/services/LectureService";
import LecturesTable from "../Lectures/LecturesTable";
import LecturesEdit from "../Lectures/LecturesEdit";
import LectureCreationPage from "../LactureCreation/LectureCreationPage";
import { ToastContainer, toast } from "react-toastify";
import WarningPopup from "../../components/warningPopup/WarningPopup";

const LecturerDashboard = () => {
  const [showCreateLecturePopup, setShowCreateLecturePopup] = useState(false);
  const [showUpdateLecturePopup, setShowUpdateLecturePopup] = useState(false);
  const [showDeleteLecturePopup, setShowDeleteLecturePopup] = useState(false);
  const [eventLectureList, setEventLectureList] = useState([]);
  const [selectedLecture, setSelectedLecture] = useState(null);
  const [selectedEventLecture, setSelectedEventLecture] = useState(null);
  const [searchLecture, setSearchLecture] = useState("");
  const [selectTable, setSelectTable] = useState("Lectures");
  const [title, setTitle] = useState("Lecture");
  const [venuesList, setVenuesList] = useState([]);
  const user = useSelector(selectUser);
  const { userId } = user || {};

  const [lectureList, setLectureList] = useState([]);

  const handleReloadEventLectureList = async () => {
    EventService.getEventByUserID(userId)
      .then((list) => {
        setEventLectureList(list);
        console.log(eventLectureList);
      })
      .catch((error) => {
        console.log("Error in getting lecture list ", error);
      })
      .finally(() => {});
  };
  //=====================================================================================
  const [showEditLectureWindow, setShowEditLectureWindow] = useState(false);
  const [editLecture, setEditLecture] = useState(null);

  const handleReloadLectureList = async () => {
    const response = await LectureService.getAllLecturesByUserId(userId);
    if (response) {
      setLectureList(response);
      console.log("lecture list : ", response);
    }
  };

  const handleEditLecture = (lecture) => {
    if (lecture) {
      setEditLecture(lecture);
      setShowEditLectureWindow(true);
    }
  };

  const handleShowDeleteWindow = () => {
    if (selectedLecture === null) {
      toast.error("Please Select Lecture To Delete.");
      return;
    }
    setShowDeleteLecturePopup(true);
  };

  const handleDeleteLecture = async () => {
    const response = await LectureService.deleteLecture(
      selectedLecture.lectureId
    );
    if (response.status === 200) {
      handleReloadLectureList();
      setSelectedLecture(null);
      toast.success(
        `Successfully ${selectedLecture.lectureName} Lecture Deleted.`
      );
      setShowDeleteLecturePopup(false);
    } else if (response.status === 400) {
      toast.error(response.data);
    } else {
      toast.error(`Error In Deleting Lecture ${selectedLecture.lectureName}.`);
    }
  };

  //=====================================================================================

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    handleReloadEventLectureList();
    handleGetLocationNameList();
    handleReloadLectureList();
  }, []);

  const handleDeleteEventLecture = async () => {
    if (selectedEventLecture) {
      EventService.deleteEvent(selectedEventLecture.eventId).then(() => {
        console.log("deleted successfully ");
        handleReloadEventLectureList();
        setSelectedEventLecture(null);
      });
    }
  };

  const handleEventLectureClick = (lecture) => {
    setSelectedEventLecture(lecture);
  };

  const handleChange = (e) => {
    setSelectTable(e.target.value);
  };

  return (
    <div className="admin-Dash">
      <ToastContainer />
      <p className="mainHead">{"Lecturer Dashboard"}</p>
      <div className="mainInform">
        {/* <TotalCount
          total={"08"}
          countIcon={
            <FaSchool
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Departments"}
        />
        <TotalCount
          total={"350"}
          countIcon={
            <IoNewspaperSharp
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Modules"}
        /> */}
      </div>
      <div className="SearchEvent">
        <select
          style={{ border: "0px", cursor: "pointer" }}
          value={selectTable}
          onChange={(e) => {
            handleChange(e);
          }}
          className="mainHead"
        >
          <option value="Lectures">Lectures List</option>
          <option value="Events">Events List</option>
        </select>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: "150px",
            padding: "3px 40px",
            border: "0.5px solid black",
            borderRadius: "5px",
            textAlign: "center",
          }}
          onChange={(e) => setSearchLecture(e.target.value)}
        />
      </div>
      {selectTable === "Lectures" ? (
        <div className="lecture-dashboard-container">
          <div className="lecture-dashboard-table">
            <LecturesTable
              lecturesList={lectureList}
              search={searchLecture}
              onLectureClick={(e) => {
                setSelectedLecture(e);
              }}
              handleLectureUpdate={(lecture) => {
                handleEditLecture(lecture);
              }}
            />
          </div>

          <div className="List-Buttons">
            <NormalButton
              handleClick={() => setShowCreateLecturePopup(true)}
              title={"Create Lecture"}
              titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
            />
            <NormalButton
              title={"Delete"}
              handleClick={handleShowDeleteWindow}
              titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
            />
          </div>
        </div>
      ) : (
        <div className="lecture-dashboard-event-table">
          <div className="lecture-dashboard-table">
            <LectureTable
              search={searchLecture}
              handleUpdateLecture={() => setShowUpdateLecturePopup(true)}
              onLectureClick={handleEventLectureClick}
              lectureList={eventLectureList} // Pass the eventList prop here
            />
          </div>

          <div className="List-Buttons">
            <NormalButton
              handleClick={() => setShowCreateLecturePopup(true)}
              title={"Create Event"}
              titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
            />
            <NormalButton
              title={"Delete"}
              handleClick={handleDeleteEventLecture}
              titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
            />
          </div>
        </div>
      )}
      {showCreateLecturePopup && (
        <div
          handleClick={() => setShowCreateLecturePopup(false)}
          className="Create-Lecture-Window"
        >
          {selectTable === "Events" && (
            <LectureCreation
              handleCloseCreateLectureWindow={() =>
                setShowCreateLecturePopup(false)
              }
              reloadLectureList={handleReloadEventLectureList}
              hideCloseButton={false}
              locationNameList={venuesList}
            />
          )}
          {selectTable === "Lectures" && (
            <LectureCreationPage
              handleCloseCreateLectureWindow={() =>
                setShowCreateLecturePopup(false)
              }
              handleReloadLectureList={handleReloadLectureList}
              hideCloseButton={false}
            />
          )}
        </div>
      )}
      {showUpdateLecturePopup && (
        <div
          handleClick={() => setShowUpdateLecturePopup(false)}
          className="Admin-Create-Event-Dashboard"
        >
          <AdminUpdateEvent
            handlecloseCreateEventWindow={() =>
              setShowUpdateLecturePopup(false)
            }
            selectedEvent={selectedEventLecture}
            reloadEventList={handleReloadEventLectureList}
            locationNameList={venuesList}
          />
        </div>
      )}
      {showEditLectureWindow && (
        <div className="Admin-Create-Event-Dashboard">
          <LecturesEdit
            selectedLecture={editLecture}
            handleCloseUpdateLectureWindow={() => {
              setShowEditLectureWindow(false);
              setEditLecture(null);
            }}
            handleReload={handleReloadLectureList}
            locationNameList={venuesList}
            reloadLecturesList={handleReloadLectureList}
          />
        </div>
      )}
      {showDeleteLecturePopup && (
        <div className="delete-lecture-container">
          <WarningPopup
            handleCloseWarningWindow={() => setShowDeleteLecturePopup(false)}
            handleOk={handleDeleteLecture}
            titleText={"Are You Sure You Want To Delete This Lecture?"}
            buttonText={"Delete"}
          />
        </div>
      )}
    </div>
  );
};

export default LecturerDashboard;
