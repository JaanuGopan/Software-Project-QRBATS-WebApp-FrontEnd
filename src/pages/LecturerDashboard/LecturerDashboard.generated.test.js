import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import LectureTable from "./LectureTable";
import TotalCount from "../../components/layout/AdminDashboardComponent/TotalCount";
import Select from 'react-select'
import { FaSchool } from "react-icons/fa6";
import { IoNewspaperSharp } from "react-icons/io5";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import LectureCreation from "./LectureCreation";
import DeleteEventService from "../../api/services/DeleteEventService";
import EventService from "../../api/services/EventService";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import AdminUpdateEvent from "../Event/AdminUpdateEvent";
import LecturerDashboard from "./LecturerDashboard";

jest.mock("../AdminDashboard/AdminDashboard.css");
jest.mock("./LectureTable");
jest.mock("../../components/layout/AdminDashboardComponent/TotalCount");
jest.mock('react-select');
jest.mock("react-icons/fa6");
jest.mock("react-icons/io5");
jest.mock("../../components/layout/AdminDashboardComponent/NormalButton");
jest.mock("react-icons/md");
jest.mock("react-icons/ri");
jest.mock("./LectureCreation");
jest.mock("../../api/services/DeleteEventService");
jest.mock("../../api/services/EventService");
jest.mock("react-redux");
jest.mock("../../redux/features/userSlice");
jest.mock("../Event/AdminUpdateEvent");

const renderTree = tree => renderer.create(tree);
describe('<LecturerDashboard>', () => {
  it('should render component', () => {
    expect(renderTree(<LecturerDashboard 
    />).toJSON()).toMatchSnapshot();
  });
  
});