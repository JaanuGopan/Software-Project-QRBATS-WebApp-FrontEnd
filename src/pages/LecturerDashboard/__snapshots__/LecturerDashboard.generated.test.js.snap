// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LecturerDashboard> should render component 1`] = `
<div
  className="admin-Dash"
>
  <p
    className="mainHead"
  >
    Staff Dashboard
  </p>
  <div
    className="mainInform"
  />
  <div
    className="SearchEvent"
  >
    <select
      className="mainHead"
      onChange={[Function]}
      style={
        Object {
          "border": "0px",
          "cursor": "pointer",
        }
      }
      value="Events"
    >
      <option
        value="Events"
      >
        Upcoming Events
      </option>
      <option
        value="Lectures"
      >
        Upcoming Lectures
      </option>
    </select>
    <input
      onChange={[Function]}
      placeholder="Search..."
      style={
        Object {
          "border": "0.5px solid black",
          "borderRadius": "5px",
          "padding": "3px 40px",
          "textAlign": "center",
          "width": "150px",
        }
      }
      type="text"
    />
  </div>
  <div
    className="AdminEventList"
  >
    <div
      className="List-Buttons"
    />
  </div>
</div>
`;
